lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,62 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(const std::uint8_t device_address,
+                         const std::uint8_t channel,
+                         hyped::io::I2c &i2c,
+                         hyped::core::ILogger &log)
+    : log_(log),
+      i2c_(i2c),
+      channel_(channel),
+      device_address_(device_address)
+{
+}
+
+Temperature::~Temperature()
+{
+}
+
+std::optional<std::int16_t> Temperature::read()
```

Comment:
sorry to bring this up again: can the sensor correctly measure negative temperatures?

if yes, then this is fine

if a negative temperature always implies that something has gone wrong (without making the assumption that there won't be negative temps outside!) then this should return `std::uint16_t` and `std::nullopt` in case of this error.

Reply:
yes it can

Reply:
And what is the scaling of the value here? Is it degrees Celsius in ints? That doesn't seem very precise...

Reply:
It is degrees Celcius. We agreed on ints since having decimal precision does not really matter when we are at 80.1 C or -10.5 C if we simply checking if temp is above or below some fixed value.

Reply:
Hm, but the sensor is more precise? I feel like it's kind of pointless to throw away that precision here. It won't give any measurable acceleration over 32-bit floats and since we're not fussed about precision anyways, the usual float caveats don't apply.

I don't feel super strongly about this, but thought I'd mention it.

Reply:
I'll ask the other teams today again if they care at all about temperature precision jic - though from our prespective it invovles a minor change that does not really affect our checks. 

@kshxtij any thoughts?

Reply:
None of the hardware teams need decimal places, so i don't really think it matters